staging/src/k8s.io/kms/service/service.go

```diff
@@ -0,0 +1,116 @@
+package service
+
+import (
+	"context"
+	"encoding/base64"
+
+	"k8s.io/klog/v2"
+	api "k8s.io/kms/apis/v2alpha1"
+	"k8s.io/kms/encryption"
+)
+
+const (
+	version              = "v2alpha1"
+	encryptedLocalKEKKey = "encryptedLocalKEK"
+)
+
+// Service offers encryption and decryption cache upfront of an remote KMS.
+type Service struct {
+	managedKeys *encryption.ManagedCipher
+}
+
+var (
+	_ api.KeyManagementServiceServer = (*Service)(nil)
+)
+
+// NewKeyManagementService creates an v2alpha1.KeyManagementServiceServer that
+// can be used for encryption and decryption, if given an remote encryption
+// service (remote KMS).
+func NewKeyManagementService(remoteCipher encryption.EncrypterDecrypter) (api.KeyManagementServiceServer, error) {
+	mk, err := encryption.NewManagedCipher(remoteCipher)
+	if err != nil {
+		klog.Infof("create key management service: %w", err)
+		return nil, err
+	}
+
+	klog.Infof("new key management service created")
+
+	return &Service{
+		managedKeys: mk,
+	}, nil
+}
+
+// Status returns version data to verify the state of the service.
+func (s *Service) Status(ctx context.Context, _ *api.StatusRequest) (*api.StatusResponse, error) {
+	return &api.StatusResponse{
+		Version: version,
+		Healthz: "ok",
+		KeyId:   base64.StdEncoding.EncodeToString(s.managedKeys.CurrentKeyID()),
```

Comment:
The `keyID` in the `StatusResponse` needs to be the remote KMS key ID and **not** the local KEK ID